portal/views/portal.py

```diff
@@ -439,16 +461,20 @@ def profile(user_id):
 @portal.route('/legal')
 def legal():
     """ Legal/terms of use page"""
+    gil = current_app.config.get('GIL', None)
     response = requests.get(app_text(LegalATMA.name_key()))
-    return render_template('legal.html', content=response.text)
+    user = current_user()
+    return render_template('legal.html' if not gil else 'gil/legal.html', content=response.text, user = user if user else None)
```

Comment:
current_user() returns None if there isn't one.  therefore you can just call:

`return render_template(..., user=curren_user())`